src/pages/index.js

```diff
@@ -58,41 +148,66 @@ addCardButton.addEventListener("click", () => {
   newCardPopup.open();
 });
 
-// Initialize an instance
-newCardPopup.setEventListeners();
-
-// UserInfo popup
-// Comment to the reviewer - I changed here the logics to use
-// the PopupWithFormClass for Profile Data change
-// and removed the extension of the popup class to the UserInfo class
-
+// UserInfo Popup
 const userPopupForm = new PopupWithForm({
   popupSelector: "#profileChange",
   handleFormSubmit: (userInput) => {
-    const newUserInput = userInput;
-    userInfo.setUserInfo({
-      name: newUserInput.name,
-      description: newUserInput.description,
-    });
-    userPopupForm.close();
+    userPopupForm.renderLoading(true);
```

Comment:
**COULD BE IMPROVED**

If it’s interesting for you here is how we can make a universal function for handling any submit. We can get rid of such duplicating as **loading effect**, **closing** and **catching errors**


```js

// You can make a universal function that accepts a request, popup instance and optional loading text  
function handleSubmit(request, popupInstance, loadingText = "Saving...") {

  // here we change the button text
  popupInstance.renderLoading(true, loadingText);
  request()
    .then(() => {
      // We need to close only in `then`
      popupInstance.close()
    })
      // we need to catch possible errors
      // console.error is used to handle errors if you don’t have any other ways for that
     .catch(console.error)
    // in `finally` we need to return the initial button text back in any case
    .finally(() => {
      popupInstance.renderLoading(false);
    });
}



```


You need to pass the **request** into the call of `handleSubmit` like that:



```js

// here is an example of the profile form handling
function handleProfileFormSubmit(inputValues) {
  // we create a function that returns a promise
  function makeRequest() {
    // `return` lets us use a promise chain `then, catch, finally` inside `handleSubmit`
    return api.editProfile(inputValues).then((userData) => {
      userInfo.setUserInfo(userData)
    });
  }
  // Here we call the function passing the request, popup instance and if we need some other loading text we can pass it as the 3rd argument
  handleSubmit(makeRequest, profilePopup);
}



```

Reply:
The function looks very cool.
Hope, one day I'll be able to write such logics myself.